FirebaseAppCheck/Sources/Core/Errors/FIRAppCheckHTTPError.m

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import "FirebaseAppCheck/Sources/Core/Errors/FIRAppCheckHTTPError.h"
+
+#import "FirebaseAppCheck/Sources/Core/Errors/FIRAppCheckErrorUtil.h"
+
+@implementation FIRAppCheckHTTPError
+
+- (instancetype)initWithHTTPResponse:(NSHTTPURLResponse *)HTTPResponse
+                                data:(nullable NSData *)data {
+  NSDictionary *userInfo = [[self class] userInfoWithHTTPResponse:HTTPResponse data:data];
+  self = [super initWithDomain:kFIRAppCheckErrorDomain
+                          code:FIRAppCheckErrorCodeUnknown
+                      userInfo:userInfo];
+  if (self) {
+    _HTTPResponse = HTTPResponse;
+    _data = data;
```

Comment:
Should we be  using the `userInfo` property of the error and providing a key for `data` instead of having a subclass here? Or is there something I'm missing for why this should be its own class?

Reply:
Working with specific NSError subclasses is so much easier in the client code because we need to check only the class to identify the type of the error and the data associated with it. In contrast we would need to check the error domain, the keys and the value types in the user info dict if we use NSError. Does it look reasonable to you?

Reply:
I think that sounds reasonable, thanks!